src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -3148,7 +3148,7 @@ bool LibraryCallKit::inline_index_partially_in_upper_range() {
   }
 
   Node* offset = argument(3);
-  Node* limit = argument(5);
+  Node* limit = argument(4);
```

Comment:
The offset is of long type so will take 2 spots (3 and 4) of argument. So limit will be argument(5).  The original code (limit = argument(5)) looks correct to me.

Reply:
@sviswa7 is right. @Hamlin-Li you do you have a test case where the value is wrong?

Reply:
@sviswa7 Oh, thanks for correct me, I did not realise this.

@vnkozlov I did run the tests `test/jdk/jdk/incubator/vector` and `test/hotspot/jtreg/compiler/vectorapi/` after applying the patch. I thought it's because this intrinsic is not covered yet, but seems I'm wrong.

I will close this pr and bug later.